src/core/gosling-component.tsx

```diff
@@ -96,63 +98,66 @@ export const GoslingComponent = forwardRef<GoslingRef, GoslingCompProps>((props,
     );
 
     // TODO: add a `force` parameter since changing `linkingId` might not update vis
-    const compile = useCallback(() => {
-        if (props.spec) {
-            const valid = gosling.validateGoslingSpec(props.spec);
+    const compile = useCallback(
+        (spec: gosling.GoslingSpec | undefined) => {
```

Comment:
I see most of the time just the `props.spec` is used here. I think this would be more understandable if we just use the `props.spec` unless an alternative spec is passed.

```suggestion
        (altSpec?: gosling.GoslingSpec) => {
                const spec = altSpec ?? props.spec;
```

(used `?` instead of `| undefined` as we used it more frequently)

and then

```diff
- compile(props.spec);
+ compile();
```

Reply:
Just changed, thanks for the suggestion!